cmd/ebrelayer/relayer/cosmos.go

```diff
@@ -0,0 +1,97 @@
+package relayer
+
+import (
+	"context"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/ethereum/go-ethereum/common"
+	tmLog "github.com/tendermint/tendermint/libs/log"
+	tmclient "github.com/tendermint/tendermint/rpc/client"
+	tmtypes "github.com/tendermint/tendermint/types"
+
+	"github.com/cosmos/peggy/cmd/ebrelayer/events"
+	"github.com/cosmos/peggy/cmd/ebrelayer/txs"
+)
+
+// InitCosmosRelayer : initializes a relayer which witnesses events on the Cosmos network and relays them to Ethereum
+func InitCosmosRelayer(tendermintProvider string, web3Provider string, bridgeContractAddress common.Address, rawPrivateKey string) error {
+	logger := tmLog.NewTMLogger(tmLog.NewSyncWriter(os.Stdout))
+	client := tmclient.NewHTTP(tendermintProvider, "/websocket")
+
+	client.SetLogger(logger)
+
+	err := client.Start()
+	if err != nil {
+		logger.Error("Failed to start a client", "err", err)
+		os.Exit(1)
+	}
+
+	defer client.Stop()
+
+	// Subscribe to all tendermint transactions
+	query := "tm.event = 'Tx'"
+
+	out, err := client.Subscribe(context.Background(), "test", query, 1000)
+	if err != nil {
+		logger.Error("Failed to subscribe to query", "err", err, "query", query)
+		os.Exit(1)
+	}
+
+	quit := make(chan os.Signal, 1)
+	signal.Notify(quit, os.Interrupt, syscall.SIGTERM)
+
+	for {
+		select {
+		case result := <-out:
+			tx, ok := result.Data.(tmtypes.EventDataTx)
+			if !ok {
+				logger.Error("Type casting failed while extracting event data from new tx")
+			}
+
+			logger.Info("\t New transaction witnessed")
+
+			txRes := tx.Result
+			for i := 1; i < len(txRes.Events); i++ {
+				event := txRes.Events[i]
+
+				var claimType events.Event
+
+				// Parse event type from event name
+				if string(event.Type) == "burn" {
+					claimType = events.MsgBurn
+				} else if string(event.Type) == "lock" {
+					claimType = events.MsgLock
+				} else {
+					claimType = events.Unsupported
+				}
+
+				switch claimType {
+				case events.MsgBurn, events.MsgLock:
```

Comment:
wrap the indented code below into a private function

Reply:
Added function `handleBurnLockMsg(...)` in https://github.com/cosmos/peggy/pull/77/commits/be4c43930db5f73c577d2a55782bf464405dd549. It uses new function `BurnLockEventToCosmosMsg()` which parses a Burn/Lock event's attribute values so that `NewCosmosMsg` is cleaner and doesn't contain any type casting.

Reply:
Might make sense for you to review this section again and confirm everything is alright as I made a lot of changes to it in https://github.com/cosmos/peggy/pull/77/commits/be4c43930db5f73c577d2a55782bf464405dd549. Thanks!